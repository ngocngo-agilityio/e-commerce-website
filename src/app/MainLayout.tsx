import { ReactNode } from 'react';

// Apis
import { getCartItems } from '@apis';

// Components
import { Header, Footer, CartStateManager } from '@components';

interface Props {
  children: ReactNode;
}

export const MainLayout = async ({ children }: Props) => {
  const { data: cartItems } = await getCartItems();

  return (
    <>
      <Header />
      {children}
      <Footer />
      <CartStateManager cartItems={cartItems} />
    </>
  );
};

export default MainLayout;
