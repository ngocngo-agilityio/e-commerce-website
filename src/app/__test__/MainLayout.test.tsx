import { render } from '@testing-library/react';

// Components
import MainLayout from '../MainLayout';

jest.mock('@components', () => ({
  Header: () => <div>Header</div>,
  Footer: () => <div>Footer</div>,
}));

describe('MainLayout', () => {
  it('should match snapshot', () => {
    const { container } = render(
      <MainLayout>
        <div>Children component</div>
      </MainLayout>,
    );

    expect(container).toMatchSnapshot();
  });
});
