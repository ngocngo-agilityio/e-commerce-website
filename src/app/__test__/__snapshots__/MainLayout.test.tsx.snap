// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainLayout should match snapshot 1`] = `
<div>
  <div>
    Header
  </div>
  <div>
    Children component
  </div>
  <div>
    Footer
  </div>
</div>
`;
